src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,165 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+
+    private static final Object[] DEFAULTCAPACITY_EMPTY_ELEMENTDATA = {};
+    private transient Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = DEFAULTCAPACITY_EMPTY_ELEMENTDATA;
+    }
+
     @Override
     public void add(T value) {
+        add(value, elementData, size);
+    }
 
+    private void add(T e, Object[] elementData, int s) {
+        if (s == elementData.length) {
+            elementData = grow();
+        }
+        elementData[s] = e;
+        size = s + 1;
     }
 
     @Override
     public void add(T value, int index) {
+        rangeCheckForAdd(index);
+        final int s;
+        Object[] elementData;
+        if ((s = size) == (elementData = this.elementData).length) {
+            elementData = grow();
+        }
+        System.arraycopy(elementData, index,
+                elementData, index + 1,
+                s - index);
+        elementData[index] = value;
+        size = s + 1;
 
     }
 
     @Override
     public void addAll(List<T> list) {
 
+        Object[] a = new Object[list.size()];
+        for (int i = 0; i < list.size(); i++) {
+            a[i] = list.get(i);
+        }
+        int numNew = a.length;
+        if (numNew == 0) {
+            return;
+        }
+        Object[] elementData;
+        final int s;
+        if (numNew > (elementData = this.elementData).length - (s = size)) {
+            elementData = grow(s + numNew);
+        }
+        int numMoved = s - 1;
+        if (numMoved > 0) {
+            System.arraycopy(elementData, size - 1,
+                    elementData, numNew,
+                    numMoved);
+        }
+        System.arraycopy(a, 0, elementData, size, numNew);
+        size = s + numNew;
```

Comment:
It can be done much easier. You already have "add" method. Use it
```suggestion
        for (int i = 0; i < list.size(); i++) {
            add(list.get(i));
        }
```